subsys/logging/backends/log_backend_adsp_hda.c

```diff
@@ -342,20 +344,20 @@ void adsp_hda_log_cavstool_hook(uint32_t written)
 	/* We *must* send this, but we may be in a timer ISR, so we are
 	 * forced into a retry loop without timeouts and such.
 	 */
-	bool done = false;
+	int ret = -1;
 
 	/*  Send IPC message notifying log data has been written */
 	do {
-		done = intel_adsp_ipc_send_message(INTEL_ADSP_IPC_HOST_DEV, IPCCMD_HDA_PRINT,
+		ret = intel_adsp_ipc_send_message(INTEL_ADSP_IPC_HOST_DEV, IPCCMD_HDA_PRINT,
 					     (written << 8) | CHANNEL);
-	} while (!done);
-
+		if (ret == -ECANCELED) {
+			return;
+		}
+	} while (ret);
```

Comment:
There is a bug here ... you changed `intel_adsp_ipc_send_message` signature but in success it is still returning `true`, so you will loop forever in case the message is sent.